node/flatpak-node-generator.py

```diff
@@ -761,19 +834,13 @@ def _handle_gulp_atom_electron(self, package: Package) -> None:
             self.gen.add_command(f'ln -sfTr "{self.electron_cache_dir}" "{cache_path}"')
 
     async def _handle_electron_headers(self, package: Package) -> None:
+        provider = NodeHeadersProvider(self.gen)
```

Comment:
Might be a bit cleaner to pass this as a constructor argument, like we do with ManifestGenerator

Reply:
Can you elaborate on this a bit, please?

Reply:
So, `ManifestGenerator` is provided as an argument to the constructor by the caller. There isn't really a major reason `NodeHeadersProvider` couldn't be the same way; it would be a bit cleaner and, most particularly, the main function is creating an instance *anyway*, might as well reuse it.

Reply:
Hm, maybe remove `NodeHeadersProvider` altogether and move it's methods to `SpecialSourceProvider`?

Reply:
I've removed `NodeHeadersProvider` altogether and added its methods to `SpecialSourceProvider`. Seems to fit well there.